### ISSUE TITLE
layout_as_tree does not check for empty graph.

### ISSUE THREAD

**Issue by clpippel** [igraph/rigraph#558]
**Describe the bug**
layout_as_tree does not check for empty graph.

**To reproduce**
`make_empty_graph() |> plot(layout=layout_as_tree)`

produces warning:
```
Warning message:
In max(res[, 2]) : no non-missing arguments to max; returning -Inf
```
If you are confident that the issue is not in the R interface but in the
C core of igraph, please add it to the main [igraph repo](https://github.com/igraph/igraph)
instead.

If you are unsure, feel free to add your issue here - we will transfer it to
the main [igraph repo](https://github.com/igraph/igraph) if the root cause is
in the C core of igraph.

**Version information**
`[1] igraph_1.3.4`

**Comment by clpippel** [comment]
The functions below also produce warnings:
```
g <- make_empty_graph()

g |> plot(layout=component_wise)
g |> plot(layout=layout_as_bipartite)
g |> plot(layout=layout_as_star)
g |> plot(layout=layout_with_gem)
g |> plot(layout=component_wise)
g |> plot(layout=layout_as_bipartite)
g |> plot(layout=layout_as_star)
g |> plot(layout=layout_with_kk)
g |> plot(layout=layout_with_lgl)
g |> plot(layout=layout_with_mds)
g |> plot(layout=layout_with_sugiyama)
g |> plot(layout=layout_)
g |> plot(layout=merge_coords)
g |> plot(layout=norm_coords)
g |> plot(layout=normalize)
```

**Comment by ntamas** [comment]
I've updated your comment above to remove redundant (duplicate) entries. As for the rest:

* `component_wise()` and `normalize()` are layout _modifiers_, not layouts, so they are not meant to be used this way.
* `merge_coords()`, `norm_coords()` and `layout_()` are not layout functions either.
* `layout_as_bipartite()` prints an _error_ saying that the graph is not bipartite, which is correct (it needs a `type` vertex attribute, even if the graph has no vertices).
* `layout_as_star()` prints an error, which is in some sense correct as you need a center vertex for a star layout and there are no vertices, so we cannot produce a star layout. Nevertheless, we should probably just special-case it and return an empty matrix.
* `layout_as_tree()`, `layout_with_gem()`, `layout_with_kk()` and `layout_with_sugiyama()` are fixed in 8bdb4627
* `layout_with_lgl()` and `layout_with_mds()` have to be fixed in the C core; I'll link the commit here once that's done.

**Comment by ntamas** [comment]
3bd98f43 fixes `layout_as_star()`, updates the C core to fix `layout_with_lgl()` and `layout_with_mds()`, and adds test to validate that these functions print no warnings for a null graph.
